README.md

```diff
@@ -14,8 +10,279 @@ But we need have a clear way to discover how you've been invited to a group with
+        </div>
+      </details>
+
+```mermaid
+graph TB
+
+root(root)
+v1(v1)
+4(4)
+d(d)
+group-invite(group-invite):::groupInvite
+helsinki(helsinki):::group
+
+subgraph Staltz
+  root --> v1 --> 4 --> helsinki
+           v1 --> d --> group-invite
+end
+
+classDef default stroke:none;
+classDef groupInvite fill: #BF2669, stroke:none, color:white;
+classDef group fill: #702A8C, stroke: none, color:white;
+```
+_Diagram showing Staltz feed state from his perspective_
+
+### 2. Group creator invites someone
+
+Staltz wants to invite his friend Arj to the group he set up, so he publishes a `group/add-member` message (which conatins the `groupKey`) to his "group-invite" feed.
+
+When Arj next stats up his application and replicates Staltz's feed (they are friends) he discovers the new `group/add-member` for him Staltz's "group-invite" feed (peers MUST replicate friends group-invite feeds).
+
+```mermaid
+graph TB
+
+rootA(root)
+v1A(v1)
+4A(4):::unreplicated
+dA(d)
+group-inviteA(group-invite):::groupInvite
+helsinkiA(helsinki):::unreplicated
+
+rootB(root)
+v1B(v1)
+9B(9)
+group-inviteB(group-invite):::groupInvite
+
+subgraph Staltz
+  rootA --> v1A --> 4A --> helsinkiA
+            v1A --> dA --> group-inviteA
+end
+
+subgraph Arj
+  rootB --> v1B --> 9B --> group-inviteB
+end
+
+classDef default stroke:none;
+classDef groupInvite fill: #BF2669, stroke:none, color:white;
+classDef group fill: #702A8C, stroke: none, color:white;
+classDef unreplicated opacity: 0.4, stroke: none;
+```
+_Diagram showing feed state of Arj and Staltz from Arj's perspective. The greyed out feeds show feeds that exist for Staltz but which Arj has yet to want to replicate_
+
+
+Assuming he accepts this invitation Arj then does the following:
+1. calculates the path to the "helsinki" group for staltz, and starts replicating it
+    - in this example `@statlz/v1/4/helsinki` (where "helsinki" is derived from the `groupKey`)
+2. creates a helsinki feed for themself
+
+
+```mermaid
+graph TB
+
+rootA(root)
+v1A(v1)
+4A(4) & dA(d)
+group-inviteA(group-invite):::groupInvite
+helsinkiA(helsinki):::group
+
+rootB(root)
+v1B(v1)
+9B(9) & cB(c)
+group-inviteB(group-invite):::groupInvite
+helsinkiB(helsinki):::group
+
+subgraph Staltz
+  rootA --> v1A --> 4A --> helsinkiA
+            v1A --> dA --> group-inviteA
+end
+
+subgraph Arj
+  rootB --> v1B --> 9B --> group-inviteB
+            v1B --> cB --> helsinkiB
+end
+
+classDef default stroke:none;
+classDef groupInvite fill: #BF2669, stroke:none, color:white;
+classDef group fill: #702A8C, stroke: none, color:white;
+```
+
+_Diagram showing updated state for Arj after he joins the group. Note the shards each feed lands in are random (but deterministic if you know the `groupKey`)_
+
+Staltz can see that Arj has accepted the invitation because he is able to decrypt the feed-announce message for Arj's "helsinki" feed, and read that the purpose is the `groupKey`.
```

Comment:
```suggestion
Staltz can see that Arj has accepted the invitation because he is able to decrypt the feed announcement message for Arj's "helsinki" feed on the shard feed, and read that the `feedpurpose` is the `groupKey`. Staltz knows which shard feed to watch for the announcement, because Arj's shard feed is deterministically determined with information Staltz is aware of.
```